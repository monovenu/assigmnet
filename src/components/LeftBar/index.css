.leftBar {
    .menu {
        width: 230px;
        height: 100vh;
        border-right:1px solid #aaa;
        .menuItem {
            cursor: pointer;
            border-bottom:1px solid #aaa;
            height: 40px;
            line-height: 40px;
            font-size: 15px;
            font-weight: 700;
            &::before {
                content:'';
                display: inline-block;
                width: 5px;
                height: 40px;
                background: transparent;
                vertical-align: top;
            }
            i {
                display: inline-block;
                margin: 0 5px 0 10px;
                color: #555;
                vertical-align: middle;
            }
            &:hover {
                color: var(--blue);
                &::before {
                    background:var(--blue);
                }
                i {
                    color: var(--blue);
                }
            }
        }
    }
}