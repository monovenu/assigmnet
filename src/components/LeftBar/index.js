import './index.css';
import {MenuList} from '@/constants'

const LeftBar = ({changeCurrentMenu}) => {


    return <div className='leftBar'>
        <ul className='menu'>
            {
                MenuList.map(item => {
                    return  <li onClick={() => changeCurrentMenu(item)} key={item} className='menuItem'>
                        <i className="iconfont icon-ICON2_API" />
                        {item}
                    </li>

                })
            }
        </ul>
    </div>
}
export default LeftBar;

