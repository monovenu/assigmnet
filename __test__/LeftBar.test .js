
import renderer from 'react-test-renderer';
import LeftBar from '../src/components/LeftBar';

it('LeftBar renders correctly', () => {
  const component = renderer.create(
    <LeftBar changeCurrentMenu={() => {}}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});