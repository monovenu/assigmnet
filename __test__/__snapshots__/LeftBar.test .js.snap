// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftBar renders correctly 1`] = `
<div
  className="leftBar"
>
  <ul
    className="menu"
  >
    <li
      className="menuItem"
      onClick={[Function]}
    >
      <i
        className="iconfont icon-ICON2_API"
      />
      User profile
    </li>
    <li
      className="menuItem"
      onClick={[Function]}
    >
      <i
        className="iconfont icon-ICON2_API"
      />
      SDK Management
    </li>
    <li
      className="menuItem"
      onClick={[Function]}
    >
      <i
        className="iconfont icon-ICON2_API"
      />
      Dashboards
    </li>
    <li
      className="menuItem"
      onClick={[Function]}
    >
      <i
        className="iconfont icon-ICON2_API"
      />
      Terms & Conditions
    </li>
  </ul>
</div>
`;
